main.js

```diff
@@ -1,32 +1,140 @@
+        
+        const colorRed1 = randomIntFromInterval(0,255)
+        const colorGreen1 = randomIntFromInterval(0,255)
+        const colorBlue1 = randomIntFromInterval(0,255)
 
-    let diffRightLeft = rightBounceDiv-leftBounceDiv
-    let difTopBottom = bottomBounceDiv - topBounceDiv
+        const colorRed2 = randomIntFromInterval(0,255)
+        const colorGreen2 = randomIntFromInterval(0,255)
+        const colorBlue2 = randomIntFromInterval(0,255)
 
     const newDiv = document.createElement("div");
-    const circleRadius = 40
+    newDiv.classList.add("circle")
+    const circleRadius = randomIntFromInterval(40,100)
     newDiv.style.width=circleRadius+"px"
     newDiv.style.height=circleRadius+"px"
-    newDiv.classList.add("circle")
-    newDiv.style.left=(leftBounceDiv+randomIntFromInterval(0,diffRightLeft-circleRadius))+"px"
-    newDiv.style.top=(topBounceDiv+randomIntFromInterval(0,difTopBottom-circleRadius))+"px"
+    newDiv.style.backgroundImage = `linear-gradient(${gradientDirection}deg, rgba(${colorRed1},${colorGreen1},${colorBlue1}), rgba(${colorRed2},${colorGreen2},${colorBlue2}))`
 
-    console.log(newDiv.style.top)
-    bounceDiv.append(newDiv)
+    return newDiv
+    }
+    
+    return function(){
+        document.getElementById("startButton").addEventListener("click", function(){
+            if(!buttonState){
+                this.innerText = "Stop"
+                let ballsNumber = parseInt(document.getElementById("ballCountInput").value)
+        
+                let {top:topBounceDiv,left:leftBounceDiv,bottom:bottomBounceDiv,right:rightBounceDiv} = bounceDiv.getBoundingClientRect();
+        
+                let diffRightLeft = rightBounceDiv-leftBounceDiv
+                let difTopBottom = bottomBounceDiv - topBounceDiv
+        
+                for(let i=0;i<ballsNumber;i++){
 
-});
+                    let newDiv = createCircle()
+                    let xCoord = (leftBounceDiv+randomIntFromInterval(0,diffRightLeft-parseInt(newDiv.style.width)))
+                    let yCoord = (topBounceDiv+randomIntFromInterval(0,difTopBottom-parseInt(newDiv.style.width)))
+                    
+                    newDiv.style.left=xCoord+"px"
+                    newDiv.style.top=yCoord+"px"
+        
+                    circlesElements.push({
+                        dom:newDiv,
+                        x:xCoord,
+                        y:yCoord,
+                        step:randomIntFromInterval(1,10),
+                        direction:randomIntFromInterval(0,1),
+                        slope:Math.random()
+                    })
+                        bounceDiv.append(newDiv)
+                }
+                
+        
+                bounceInterval = setInterval(()=>{
+                    console.log("Bounce interval")
+                    let {top:topBounceDiv,left:leftBounceDiv,bottom:bottomBounceDiv,right:rightBounceDiv} = bounceDiv.getBoundingClientRect();
```

Comment:
I do this because the boundingRect changes when I inspect the page.

<img width="960" alt="BoundingRect" src="https://user-images.githubusercontent.com/87003862/126335044-62b0cd5c-e50c-4afe-99e7-c3c1d1eb682c.png">

Reply:
Right, so you wanted to handle the window resize use case. Good 👍